### PR TITLE
feat(swarm): remove `KeepAlive::Until` from `OneshotHandler`

### PR DESCRIPTION
**Pull request by leonzchang**
## Description

Related: #3844.
Related: #4656.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: swarm/src/handler/one_shot.rs [hunk 1]
```diff
@@ -232,8 +223,6 @@ where
 /// Configuration parameters for the `OneShotHandler`
 #[derive(Debug)]
 pub struct OneShotHandlerConfig {
-    /// Keep-alive timeout for idle connections.
-    pub keep_alive_timeout: Duration,
```
Comment (thomaseizinger) [inline swarm/src/handler/one_shot.rs:239]: Removing this field is a breaking change. Mind sending a deprecation PR for it?

**Review by thomaseizinger — COMMENTED** [review]
Awesome, thank you!

Just one comment.

File: swarm/src/handler/one_shot.rs (hunk 1 above)
Comment (leonzchang) [reply swarm/src/handler/one_shot.rs:239]: opened #4680

File: swarm/CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +21,7 @@
   See [PR 4680](https://github.com/libp2p/rust-libp2p/pull/4680).
 
 [PR 4120]: https://github.com/libp2p/rust-libp2p/pull/4120
+[PR 4677]: https://github.com/libp2p/rust-libp2p/pull/4677
```
Comment (thomaseizinger) [inline swarm/CHANGELOG.md]:
```suggestion
```

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**